src/qtils/strict_sptr.hpp

```diff
@@ -0,0 +1,99 @@
+/**
+ * Copyright Quadrivium LLC
+ * All Rights Reserved
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include <memory>
+#include <stdexcept>
+#include <type_traits>
+
+namespace qtils {
+
+  /**
+   * @brief A strict version of std::shared_ptr that guarantees non-null values.
+   *
+   * This class prevents default construction and ensures that the shared
+   * pointer is never null. It removes unsafe methods such as `reset()` and
+   * `operator=`. Provides safe access to the underlying std::shared_ptr.
+   *
+   * @tparam T The type of the managed object.
+   */
+  template <class T>
+    requires(not std::is_void_v<T>)
```

Comment:
```suggestion
```

Reply:
It works as reference, but we cannot have reference to void.